### PR TITLE
[PT] Fix typo and some mistakes in Brazilian Portuguese on pt/lessons/advanced/typespec.md

### PR DESCRIPTION
**Pull request by herminiotorres**
Fix typo and some mistakes in Brazilian Portuguese.

I decided to translate some words in Portuguese instead of staying in English for more sense and understanding and improve some phrases too.

### CODE REVIEW

File: pt/lessons/advanced/typespec.md [hunk 1]
```diff
@@ -74,9 +74,9 @@ defmodule Examples do
 end
 ```
 
-Já definimos o tipo `t(first, last)`, que é uma representação da estrutura `%Examples{first: first, last: last}`. Nesse ponto, vemos tipos que podem receber parâmetros, mas definimos o tipo `t` e nesse momento, ele é uma representação da estrutura `%Examples{first: integer, last: integer}`.   
+Já definimos o tipo `t(first, last)`, que é uma representação da estrutura `%Examples{first: first, last: last}`. Nesse ponto, vemos tipos que podem receber parâmetros, mas definimos o tipo `t` e nesse momento, ele é uma representação da estrutura `%Examples{first: integer, last: integer}`.
 
-Qual a diferença? A primeira representa a estrutura `Examples` e as duas chaves poderiam receber qualquer tipo. A segunda representa a estrutura que as chaves são `integers`. Que significa um código como este:
+Qual a diferrença ? A primeira representa a estrutura `Examples` e as duas chaves poderiam receber qualquer tipo. A segunda representa a estrutura que as chaves recebem são do tipo `integers`. Que significa um código como este:
```
Comment (vbrazo) [inline pt/lessons/advanced/typespec.md]: diferrença? I think it's just one `r`

File: pt/lessons/advanced/typespec.md [hunk 2]
```diff
@@ -54,11 +54,11 @@ def sum_times(a, params) do
 end
 ```
 
-Inserimos uma estrutura no módulo `Examples` que contém dois campos, `first` e `last`. Essa é uma versão simples de estrutura do módulo `Range`. Falaremos sobre `structs` quando iniciarmos as discussões [modules](../../basics/modules/#structs). Vamos imaginar que precisamos especificar a estrutura `Examples` em vários lugares. Seria chato escrever especificações longas, complexas e isso seria uma fonte de bugs. Uma solução para esse problema é `@type`.
+Inserimos uma estrutura no módulo `Examples` que contém dois campos, `first` e `last`. Essa é uma versão simples de estrutura do módulo `Range`. Falaremos sobre `structs` quando iniciarmos as discusões sobre [módulos](../../basics/modules/#structs). Vamos imaginar que precisamos especificar a estrutura `Examples` em vários lugares. Seria chato escrever especificações longas, complexas e isso seria uma fonte de bugs. Uma solução para esse problema é `@type`.
```
Comment (vbrazo) [inline pt/lessons/advanced/typespec.md]: Discussões has 2 `ss`

File: pt/lessons/advanced/typespec.md [hunk 3]
```diff
@@ -74,9 +74,9 @@ defmodule Examples do
 end
 ```
 
-Já definimos o tipo `t(first, last)`, que é uma representação da estrutura `%Examples{first: first, last: last}`. Nesse ponto, vemos tipos que podem receber parâmetros, mas definimos o tipo `t` e nesse momento, ele é uma representação da estrutura `%Examples{first: integer, last: integer}`.   
+Já definimos o tipo `t(first, last)`, que é uma representação da estrutura `%Examples{first: first, last: last}`. Nesse ponto, vemos tipos que podem receber parâmetros, mas definimos o tipo `t` e nesse momento, ele é uma representação da estrutura `%Examples{first: integer, last: integer}`.
 
-Qual a diferença? A primeira representa a estrutura `Examples` e as duas chaves poderiam receber qualquer tipo. A segunda representa a estrutura que as chaves são `integers`. Que significa um código como este:
+Qual a diferença ? A primeira representa a estrutura `Examples` e as duas chaves poderiam receber qualquer tipo. A segunda representa a estrutura que as chaves recebem são do tipo `integers`. Que significa um código como este:
```
Comment (vbrazo) [inline pt/lessons/advanced/typespec.md]: We don’t need this extra space before the ?

Comment (herminiotorres) [reply pt/lessons/advanced/typespec.md]: indeed I commit this change and remove the blank space, thanks a lot for your review @vbrazo

**Pull request merged**